src/spec_procs.cpp

```diff
@@ -1052,10 +1071,34 @@ SPECIAL(teacher)
     }
     lose_nuyen(ch, skill_nuyen_cost, NUYEN_OUTFLOW_SKILL_TRAINING);
   }
-  if (GET_SKILL_POINTS(ch) > 0)
-    GET_SKILL_POINTS(ch)--;
-  else
-    GET_KARMA(ch) -= get_skill_price(ch, skill_num) * 100;
+  if (skills[skill_num].requires_resonance) {
+    // otaku resonance skills are either bought with channel points or by karma, not skill points.
+    if (GET_CHANNEL_POINTS(ch) > 0) {
+        // Channel points exist only in CharGen. 
+      if (REAL_SKILL(ch, skill_num) + 1 <= 3) {
+        // We can have as many skills below or at 3 as we want with channel points
+        GET_CHANNEL_POINTS(ch)--;
+      } else {
+        // Since we're raising this skill above 3 we have to check we don't bypass the spread of 6, 5, 4, 3, 3.
+        // which is the limit for otaku skills in chargen.
+        for (int ci=154; i < 159;i++) {
+          if (REAL_SKILL(ch, ci) == REAL_SKILL(ch, skill_num) + 1) {
+            send_to_char(ch, "When buying channel skills with freebies you can only have one channel skill at %d, and %s is already at that value.",
+              REAL_SKILL(ch, skill_num) + 1, skills[ci].name);
+              return FALSE;
+          }
+        }
+        GET_CHANNEL_POINTS(ch)--;
+      }
+    } 
+    else
+      GET_KARMA(ch) -= get_skill_price(ch, skill_num) * 100;
```

Comment:
Embedding this else-statement here means folks can't use skill points on channels. Recommend changing the overall logic to:
```c++
if (skills[skill_num].requires_resonance && GET_CHANNEL_POINTS(ch) > 0) {
  //pay with chanp
} else if (GET_SKILL_POINTS(ch) > 0) {
  //pay with skillp
} else {
  //pay with karma
}